src/v2/BaseLoginRouter.js

```diff
@@ -77,25 +76,31 @@ export default Router.extend({
       settings: this.settings,
     });
 
+    configIdxJsClient(this.appState);
     this.listenTo(this.appState, 'remediationSuccess', this.handleIdxResponseSuccess);
     this.listenTo(this.appState, 'remediationError', this.handleIdxResponseFailure);
   },
 
   handleIdxResponseSuccess(idxResponse) {
     if (idxResponse.interactionCode) {
+      // Although session.stateHandle isn't used by interation flow,
+      // it's better to clean up at the end of the flow.
+      sessionStorageHelper.removeStateHandle();
       // This is the end of the IDX flow, now entering OAuth
       return interactionCodeFlow(this.settings, idxResponse);
     }
 
-    // transform response
     const lastResponse = this.appState.get('idx');
-    const ionResponse = transformIdxResponse(this.settings, idxResponse, lastResponse);
 
-    // if this is a terminal error, clear all transaction meta
-    if (IonResponseHelper.isTerminalError(ionResponse)) {
-      clearTransactionMeta(this.settings);
+    // Do not save state handle for the first page loads.
```

Comment:
can you also outline, why we should not save stateHandle for first page load.

Reply:
Added more comments.